amoro-ams/src/main/java/org/apache/amoro/server/table/TableRuntime.java

```diff
@@ -611,11 +611,14 @@ public String toString() {
   public long getQuotaTime() {
     long calculatingEndTime = System.currentTimeMillis();
     long calculatingStartTime = calculatingEndTime - AmoroServiceConstants.QUOTA_LOOK_BACK_TIME;
-    taskQuotas.removeIf(task -> task.checkExpired(calculatingStartTime));
-    long finishedTaskQuotaTime =
-        taskQuotas.stream()
-            .mapToLong(taskQuota -> taskQuota.getQuotaTime(calculatingStartTime))
-            .sum();
+    long finishedTaskQuotaTime;
+    synchronized (taskQuotas) {
```

Comment:
Can we use a concurrency-safe type instead? like `CopyOnWriteArrayList`?

Reply:
Thank you for the opportunity. I've updated the code to use CopyOnWriteArrayList.